core/src/main/scala/chisel3/internal/Builder.scala

```diff
@@ -202,6 +202,18 @@ private[chisel3] trait HasId extends InstanceId {
       setRef(Ref(available_name))
     }
 
+  /** Create a name only if a ref provided, a name is suggested, or
+    * [[autoSeed]] exists.
+    */
+  private[chisel3] def forceOptionalName(namespace: Namespace): Unit =
+    if(_ref.isEmpty) {
+      val candidate_name = computeName(None, None)
+      if (!candidate_name.isEmpty) {
+        val available_name = namespace.name(candidate_name.get)
+        setRef(Ref(available_name))
+      }
+    }
+
```

Comment:
@jackkoenig please check this implementation

Reply:
not sure if i should be providing different arguments to `computeName`

Reply:
Are you trying to allow `assert` statements to not have a name?
If so, I think we should just name them all, even if it isn't strictly necessary in order to simplify things.

Reply:
yeah, the hope is to avoid IR size issues (probably a sifive-specific problem)

Reply:
And just general clutter of naming them all, but of course we could and it would be a bit simpler.

Reply:
I don't think it would have any significant overhead since there are so few of them. If all verification statements have names, that would also open up the possibility to switch to generating SystemVerilog `property` assertions instead of concurrent assertions.
So personally I would recommend to avoid the complexity and just go all in on naming verification statements.

Reply:
ok, i'll go ahead and name everything then